tartiflette/language/validators/query/utils.py

```diff
@@ -0,0 +1,32 @@
+from tartiflette.types.helpers.reduce_type import reduce_type
+
+
+def find_field(parent_type_name, field_name, schema):
+    try:
+        return schema.type_definitions[parent_type_name].find_field(field_name)
+    except (AttributeError, KeyError):
+        return None
+
+
+def find_field_reduced_type(parent_type_name, field_name, schema):
+    schema_field = find_field(parent_type_name, field_name, schema)
+
+    if not schema_field:
+        return None
+
+    return schema.type_definitions[reduce_type(schema_field.gql_type)]
+
+
+def get_schema_field_type_name(parent_type_name, schema, field_name):
+    try:
+        return reduce_type(
+            schema.type_definitions[parent_type_name]
+            .find_field(field_name)
+            .gql_type
+        )
+    except (AttributeError, KeyError):
+        return None
+
+
+def find_nodes_named(nodes, name):
```

Comment:
I think we should find a better naming for this function. As first sight I thought it was looking for `NamedNode` nodes over the `nodes` parameter.

Reply:
what do you think of `find_nodes_by_name` ?